TradHeli: Add PreArm and sanity checking to servo_test

JoshWelsh opened the pull request:
Found in the field, current behavior will allow arming while a servo test of the swash and rudder are still happening.  If arming does occur the flight will be allowed and when the Heli lands and ultimately disarms, the servo test will resume.  If the heli still has any rotor RPM, this can be a bit devastating. Biggest thanks ever to @lthall and @bnsgeyer for the detective work and help structuring the checks.

rmackay9 commented on ArduCopter/AP_Arming.cpp:
```diff
@@ -298,6 +298,13 @@ bool AP_Arming_Copter::motor_checks(bool display_failure)
+	    //servo_test check
+#if FRAME_CONFIG == HELI_FRAME
+    if(copter.motors->servo_test_running()) {
+        check_failed(display_failure, "Servo Test is still running!");
```
Let's remove the exclamation point at the end of the comment if that's ok.

JoshWelsh replied:
Ha, I had just done that before I even saw this, thanks @rmackay9 I agree.  That was a carry-over from when it said "Don't pull a Josh!"

rmackay9 commented on libraries/AP_Motors/AP_MotorsHeli.cpp:
```diff
@@ -171,6 +171,8 @@ void AP_MotorsHeli::output()
     output_logic();
 
     if (_flags.armed) {
+	//block servo_test from happening at disarm
```
small white space issue here.   Could you indent the comment a bit more to line up with the code?  There are a few other white space issues too.

JoshWelsh replied:
Thanks, I'll have a look now

rmackay9 commented on libraries/AP_Motors/AP_MotorsHeli.h:
```diff
@@ -130,6 +130,8 @@ class AP_MotorsHeli : public AP_Motors {
     void set_enable_bailout(bool bailout) { _heliflags.enable_bailout = bailout; }
 
+    // return true if the servo test is still running/pending
+    bool servo_test_running() const { return _heliflags.servo_test_running; }
```
Let's add a blank line below this declaration

rmackay9 commented on libraries/AP_Motors/AP_MotorsHeli.h:
```diff
@@ -199,6 +201,7 @@ class AP_MotorsHeli : public AP_Motors {
         uint8_t save_rsc_mode           : 1;    // used to determine the rsc mode needs to be saved while disarmed
         uint8_t in_autorotation         : 1;    // true if aircraft is in autorotation
         uint8_t enable_bailout          : 1;    // true if allowing RSC to quickly ramp up engine
+		uint8_t servo_test_running      : 1;    // true if servo_test is running
```
couple more spaces required at the front here.  sorry to be nitpicky.

JoshWelsh replied:
Appreciate the nitpickiness!

rmackay9 commented:
Txs for this.  I've added @bnsgeyer as the review 'cuz it's really his baby.

JoshWelsh commented:
@rmackay9 @bnsgeyer I think I resolved the findings, some of these appear to be from the cherry picking in notepad++ using tabs instead of spaces, but I think I caught everything in the github editor.  Let me know if they're still not resolved on your side and I'll have another look.

bnsgeyer commented:
@rmackay9   I added this PR to Copter 4.0 backports.  Hopefully you can get this into 4.0.4.  Thanks!

JoshWelsh commented:
FYI I flew this today on the UAVCAN Heli and verified it performs as expected.

rmackay9 commented:
@bnsgeyer, @JoshWelsh, ok, i've backported it and it went in after resolving a small merge conflict.  Maybe you could check the Copter-4.0 branch is OK?  Sorry to be annoying but it's probably best if the experts double check that I didn't mess it up.

rmackay9 commented:
@JoshWelsh by the way, I fixed up a couple of white space issues. No big deal but I think your editor might produce tabs in some cases?  Some editors allow always inserting 4 spaces instead of using tab.

JoshWelsh commented:
> @JoshWelsh by the way, I fixed up a couple of white space issues. No big deal but I think your editor might produce tabs in some cases? Some editors allow always inserting 4 spaces instead of using tab.

Hmm, ok thanks for the heads up @rmackay9 I thought I had cleaned all those up but I must have missed some.  Really appreciate the assist!
